app/client/src/contexts/MapHighlight.tsx

```diff
@@ -0,0 +1,45 @@
+import { createContext, useContext, useMemo, useState } from 'react';
+import type { ReactNode } from 'react';
+import Graphic from '@arcgis/core/Graphic';
+
+type State = {
+  highlightedGraphic: Graphic | null;
+  selectedGraphic: Graphic | null;
+  setHighlightedGraphic: (graphic: Graphic | null) => void;
+  setSelectedGraphic: (graphic: Graphic | null) => void;
+};
+
+const MapHighlightContext = createContext<State | undefined>(undefined);
+
+type Props = { children: ReactNode };
+
+export function MapHighlightProvider({ children }: Props) {
+  const [highlightedGraphic, setHighlightedGraphic] = useState<Graphic | null>(
+    null,
+  );
+  const [selectedGraphic, setSelectedGraphic] = useState<Graphic | null>(null);
+  const context: State = useMemo(() => {
+    return {
+      highlightedGraphic,
+      setHighlightedGraphic,
+      selectedGraphic,
+      setSelectedGraphic,
+    };
+  }, [highlightedGraphic, selectedGraphic]);
+
+  return (
+    <MapHighlightContext.Provider value={context}>
+      {children}
+    </MapHighlightContext.Provider>
+  );
+}
+
+export function useMapHighlightContext() {
```

Comment:
Similar comments as those on the `FullscreenContext`. Is there a specific reason why this context is being treated as a custom hook?

I feel like this should follow the same pattern as the `AddDataWidget.tsx` context file for consistency.

I tested switching it over and it seems like everything is still working. If we switch we also don't need to use `useMemo` either.